static/js/components/GroupPageManageUserButtons.jsx

```diff
@@ -63,49 +76,88 @@ const ManageUserButtons = ({ group, loadedId, user, isAdmin }) => {
     }
   };
 
+  const handleDelete = () => {
+    dispatch(
+      groupsActions.deleteGroupUser({
+        userID: user.id,
+        group_id: group.id,
+      })
+    );
+    history.goBack();
+  };
+
   return (
     <div>
-      <Button
-        size="small"
-        onClick={() => {
-          toggleUserAdmin(user);
-        }}
-        disabled={isAdmin(user) && numAdmins === 1}
-      >
-        <span style={{ whiteSpace: "nowrap" }}>
-          {isAdmin(user) ? "Revoke admin status" : "Grant admin status"}
-        </span>
-      </Button>
-      &nbsp;|&nbsp;
-      <Tooltip title="Manage whether user can save sources to this group.">
-        <Button
-          size="small"
-          onClick={() => {
-            toggleUserCanSave(user);
-          }}
-        >
-          <span style={{ whiteSpace: "nowrap" }}>
-            {canSave(user) ? "Revoke save access" : "Grant save access"}
-          </span>
-        </Button>
-      </Tooltip>
-      &nbsp;|&nbsp;
-      <IconButton
-        edge="end"
-        aria-label="delete"
-        data-testid={`delete-${user.username}`}
-        onClick={() =>
-          dispatch(
-            groupsActions.deleteGroupUser({
-              userID: user.id,
-              group_id: group.id,
-            })
-          )
-        }
-        disabled={isAdmin(user) && numAdmins === 1}
-      >
-        <DeleteIcon />
-      </IconButton>
+      {isAdmin(currentUser) && (
+        <div>
+          <Button
+            size="small"
+            onClick={() => {
+              toggleUserAdmin(user);
+            }}
+            disabled={isAdmin(user) && numAdmins === 1}
+          >
+            <span style={{ whiteSpace: "nowrap" }}>
+              {isAdmin(user) ? "Revoke admin status" : "Grant admin status"}
+            </span>
+          </Button>
+          &nbsp;|&nbsp;
+          <Tooltip title="Manage whether user can save sources to this group.">
+            <Button
+              size="small"
+              onClick={() => {
+                toggleUserCanSave(user);
+              }}
+            >
+              <span style={{ whiteSpace: "nowrap" }}>
+                {canSave(user) ? "Revoke save access" : "Grant save access"}
+              </span>
+            </Button>
+          </Tooltip>
+          &nbsp;|&nbsp;
+        </div>
+      )}
+      {(isAdmin(currentUser) || user.username === currentUser.username) && (
+        <div>
+          <IconButton
+            edge="end"
+            aria-label="delete"
+            data-testid={`delete-${user.username}`}
+            onClick={() => setConfirmDeleteOpen(true)}
+            disabled={isAdmin(user) && numAdmins === 1}
+          >
+            <DeleteIcon />
+          </IconButton>
+          <Dialog
+            fullWidth
+            open={confirmDeleteOpen}
+            onClose={handleConfirmDeleteDialogClose}
+          >
+            <DialogTitle>Remove user?</DialogTitle>
```

Comment:
Should have a different title user:
  1) "Remove yourself?" if user.username === currentUser.username
  2) Otherwise "Remove user?"

Reply:
yes